src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java

```diff
@@ -106,6 +106,22 @@
           "LIKE", "NOT LIKE", "BETWEEN", "NOT BETWEEN", "IN", "NOT IN", "IS NULL", "IS NOT NULL",
           "IS TRUE", "IS NOT TRUE", "IS FALSE", "IS NOT FALSE");
 
+  @Override
+  public RecordWriter<StructuredRecord, NullWritable> getRecordWriter(TaskAttemptContext taskAttemptContext)
```

Comment:
The classes BigQueryRecordWriter and BigQueryAvroConverter were introduced as an abstraction layer to switch between JSON and AVRO output format at runtime. Since prior to 6.4 we only use JSON , these are unnecessary here. Use StructuredToAvroTransformer with the getOutputSchema to write directly with the existing RecordWriter.

Reply:
This PR will also introduce the problem in this bug - https://cdap.atlassian.net/browse/PLUGIN-678. Please also cherry pick the fix for that as part of this PR.

Reply:
We followed your advice. Also the fix for PLUGIN-678 is included.